core/storage/trie/polkadot_trie_db/polkadot_trie_batch.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/trie/polkadot_trie_db/polkadot_trie_batch.hpp"
+
+namespace kagome::storage::trie {
+
+  using common::Buffer;
+
+  PolkadotTrieBatch::PolkadotTrieBatch(PolkadotTrieDb &trie) : trie_{trie} {}
+
+  outcome::result<void> PolkadotTrieBatch::put(const Buffer &key,
+                                               const Buffer &value) {
+    auto value_copy = value;
+    return put(key, std::move(value_copy));
+  }
+
+  outcome::result<void> PolkadotTrieBatch::put(const Buffer &key,
+                                               Buffer &&value) {
+    if (value.empty()) {
+      OUTCOME_TRY(remove(key));
+    } else {
+      commands_.push_back({Action::PUT, key, std::move(value)});
```

Comment:
```suggestion
      commands_.emplace_back(Action::PUT, key, value);
```

Reply:
If you are going to use rvalues here, `std::forward` is needed

Reply:
emplace won't compile, seemingly has an issue with aggregate construction.